src/Microsoft.Tye.Hosting/DockerRunner.cs

```diff
@@ -294,6 +294,37 @@ private async Task StopContainerAsync(Tye.Hosting.Model.Service service)
                 : Path.Combine(root, ".microsoft", "usersecrets");
         }
 
+        public async Task HandleStaleReplica(ReplicaEvent replicaEvent)
+        {
+            var container = replicaEvent.Replica.Name;
+
+            if (container != "")
+            {
+                await ProcessUtil.RunAsync("docker", $"rm -f {container}",
+                    throwOnError: false,
+                    outputDataReceived: data => _logger.LogInformation("removed container {container} from previous run", container));
+            }
+        }
+
+        public ValueTask<IDictionary<string, string>> SerializeReplica(ReplicaEvent replicaEvent)
```

Comment:
Should this just be its own data type instead of piggy-backing on `ReplicaEvent`?

This doesn't roundtrip since we're just using the service name instead of the service definition.

Reply:
The service name of the PID if it's a process.
But yeah, most of the fields are not necessary.
I could 1) Just use a dictionary or 2) Use a string (that contains the container name or PID).
In both cases, the serialize and deserialize will become obsolete and you would only have a `HandleStaleReplica` method

Reply:
It could also just be its own type. It's a little strange to use `ReplicaEvent` because you're not going to rehydrate all of its fields, just the basic stuff like the PID. I think either using a dictionary or using a new type would both be improvments.